### PR TITLE
Series of Lite-On Light (ALS) and Proximity(PS) sensors - LTR-303, LTR-329, LTR-553, LTR-556, LTR-659, some others

### PR DESCRIPTION
**Pull request by latonita**
## Description:

Documentation for LTR303 component

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/2485

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6076

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/ltr_als_ps.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
+Lite-On Ambient Light & Proximity Sensors
+==================================================
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
Lite-On Ambient Light & Proximity Sensors
=========================================
```

File: components/sensor/ltr_als_ps.rst [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    LTR-329 and LTR-303 Ambient Light Sensors.
+
+.. figure:: images/ltr303-ui.png
+    :align: center
+    :width: 60.0%
+
+    LTR-303 Sensor in Home Assistant UI.
+
+The ``ltr_als_ps`` sensor platform allows you to use family of several LiteOn ambient light and proximity sensors
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
The ``ltr_als_ps`` sensor platform allows you to use a range of LiteOn ambient light and proximity sensors
```

File: components/sensor/ltr_als_ps.rst [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+- Integrated Ambient Light and Proximity Sensors **LTR-553ALS**, **LTR-556ALS**, **LTR-559ALS**
+- Proximity Sensor **LTR-659ALS** (with PS gain control)
+
+The sensors are very similar and share the same datasheet. First two have default I²C address ``0x29`` and the rest
+have default I²C address ``0x23``. The sensors are available on breakout boards from a number of vendors, for 
+example `Adafruit`_ or `Pimoroni`_.
+
+Proximity sensors are the same sort of sensors that you find in phones and tablets to disable the screen when you hold
+the device up to your ear. They might be useful for automated turning on or off of displays and control panels. Good
+example is CoreS3 board from M5Stack that has LTR-559ALS sensor integrated just under the screen.
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
the device up to your ear. They might be useful for automated turning on or off of displays and control panels. An
example is the CoreS3 board from M5Stack that has an LTR-559ALS sensor integrated just under the screen.
```

File: components/sensor/ltr_als_ps.rst [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+.. _Adafruit: http://www.adafruit.com/products/5610
+.. _Pimoroni: https://shop.pimoroni.com/products/ltr-559-light-proximity-sensor-breakout
+
+Ambient light sensing
+---------------------
+
+Sensor platform provides a linear response over a wide dynamic range from 0.01 lux to 64k lux and is well suited
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
The sensor platform provides a linear response over a wide dynamic range from 0.01 lux to 64k lux and is well suited 
```

File: components/sensor/ltr_als_ps.rst [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+Ambient light sensing
+---------------------
+
+Sensor platform provides a linear response over a wide dynamic range from 0.01 lux to 64k lux and is well suited 
+to applications under high ambient brightness. There are altogether six gain settings (1X, 2X, 4X, 8X, 48X and 96X)
+available for user to configure. Use higher gain for dimmer areas.
+
+Devices consists of two photodiodes: *CH0* diode that is sensitive to both visible and infrared light and 
+*CH1* diode that is sensitive only to infrared light.
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
The devices consist of two photodiodes: The *CH0* diode is sensitive to both visible and infrared light and 
the *CH1* diode is sensitive only to infrared light.
```

File: components/sensor/ltr_als_ps.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+.. figure:: images/ltr303-spectral.png
+    :align: center
+    :width: 100.0%
+
+    CH0 and CH1 channels typical spectral response
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
    CH0 and CH1 typical spectral response
```

File: components/sensor/ltr_als_ps.rst [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+Proximity sensing
+-----------------
+
+Proximity sensor has built-in emitter and detector. The sensor detects reflected IR light from the emitter and
+gives a raw count value inversely exponential to the distance. Drop in the count value means an object is getting
+further away from the sensor, and vice versa. Neither of the datasheets provides any information on how to convert
+the raw count value to distance. The only way to do it is to test the sensor yourself and select the threshold
+according to your needs and environment. Exact numbers will depend on the type of the object, its color and 
+reflectivity.
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
The proximity sensor has a built-in emitter and detector. The sensor detects reflected IR light from the emitter and
gives a raw count value inversely exponential to the distance. A decrease in the count value means an object is getting
further away from the sensor, and vice-versa. Neither of the datasheets provides any information on how to convert
the raw count value to distance. The only way to do it is to test the sensor yourself and select the threshold
according to your needs and environment. Exact numbers will depend on the type of the object, its color and 
reflectivity.
```

File: components/sensor/ltr_als_ps.rst [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+- **on_ps_low_threshold** (*Optional*): Actions to perform when the proximity sensor is triggered
+  on object getting further away.
+
+Sensors
+^^^^^^^
+This component offers five sensors for ALS-enabled devices and one sensor for PS-enabled devices.
+You can configure all or any subset of the sensors. Each configured sensor is reported separately 
+on each ``update_interval``. **name** option is required for the sensor. All other options from 
+:ref:`Sensor <config-sensor>`. If you don’t need to configure any other sensor options, you can use 
+the simplified syntax for the sensor: ``ambient_light: "Ambient light"``, for example.
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
This component offers five sensors for ALS-equipped devices and one sensor for PS-equipped devices.
You can configure all or any subset of these sensors. Each configured sensor is reported separately 
on each ``update_interval``. Each is an ESPHome :ref:`sensor <config-sensor>` and may be configured
accordingly; if you don’t need to configure additional :ref:`sensor <config-sensor>` variables, you may simply
use the shorthand syntax for the sensor. For example: ``ambient_light: "Ambient light"``
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
A few minor corrections/rephrasing

File: components/sensor/ltr_als_ps.rst [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+
+Example configuration
+---------------------
+
+.. code-block:: yaml
+
+    i2c:
+      # ...
+
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
```

File: components/sensor/ltr_als_ps.rst [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+- **on_ps_low_threshold** (*Optional*): Actions to perform when the proximity sensor is triggered
+  on object getting further away.
+
+Sensors
+^^^^^^^
+This component offers five sensors for ALS-equipped devices and one sensor for PS-equipped devices.
+You can configure all or any subset of these sensors. Each configured sensor is reported separately 
+on each ``update_interval``. Each is an ESPHome :ref:`sensor <config-sensor>` and may be configured
+accordingly; if you don’t need to configure additional :ref:`sensor <config-sensor>` variables, you may simply
+use the shorthand syntax for the sensor. For example: ``ambient_light: "Ambient light"``
```
Comment (kbx81) [inline components/sensor/ltr_als_ps.rst]:
```suggestion
accordingly; if you don’t need to configure additional :ref:`sensor <config-sensor>` variables, you
may simply use the shorthand syntax for the sensor. For example: ``ambient_light: "Ambient light"``
```

File: index.rst [hunk 11]
```diff
@@ -438,6 +438,12 @@ Light
     APDS9960, components/sensor/apds9960, apds9960.jpg, Colour & Gesture
     AS7341, components/sensor/as7341, as7341.jpg, Spectral Color Sensor
     BH1750, components/sensor/bh1750, bh1750.jpg, Lux
+    LTR329, components/sensor/ltr_als_ps, ltr329.jpg, Lux
+    LTR303, components/sensor/ltr_als_ps, ltr303.jpg, Lux
+    LTR553, components/sensor/ltr_als_ps, ltr-ps.jpg, Lux & Proximity
+    LTR556, components/sensor/ltr_als_ps, ltr-ps.jpg, Lux & Proximity
+    LTR559, components/sensor/ltr_als_ps, ltr559.jpg, Lux & Proximity
+    LTR659, components/sensor/ltr_als_ps, ltr-ps.jpg, Proximity
     LTR390, components/sensor/ltr390, ltr390.jpg, Lux & UV
```
Comment (kbx81) [inline index.rst]:
```suggestion
    LTR303, components/sensor/ltr_als_ps, ltr303.jpg, Lux
    LTR329, components/sensor/ltr_als_ps, ltr329.jpg, Lux
    LTR390, components/sensor/ltr390, ltr390.jpg, Lux & UV
    LTR553, components/sensor/ltr_als_ps, ltr-ps.jpg, Lux & Proximity
    LTR556, components/sensor/ltr_als_ps, ltr-ps.jpg, Lux & Proximity
    LTR559, components/sensor/ltr_als_ps, ltr559.jpg, Lux & Proximity
    LTR659, components/sensor/ltr_als_ps, ltr-ps.jpg, Proximity
```

**Review by kbx81 — CHANGES_REQUESTED** [review]

**Comment by latonita** [comment]
@kbx81 thanks for the updates

**Review by kbx81 — APPROVED** [review]

**Pull request merged**